packages/perseus/src/widgets/radio/utils/hash-shuffle.ts

```diff
@@ -0,0 +1,106 @@
+import {generateChoiceId} from "@khanacademy/perseus-core";
+
+import type {RadioChoiceWithMetadata} from "../radio.class.new";
+
+/**
+ * MumurHash3 32-bit implementation
+ * Implementation reference: https://mojoauth.com/hashing/murmurhash-in-typescript
+ * Optimized for string inputs without external dependencies
+ */
+function hashString(key: string): number {
```

Comment:
Same as the other PR, is there a reason we can't leverage someone else's work on the algorithm? Some small npm package or something? I'm not smart enough to understand bit shifting and if it's in the codebase I'll inevitably have to look at it.